test/cypress/integration/group-paying.spec.js

```diff
@@ -0,0 +1,160 @@
+
+    cy.getByDT('recordPayment').click()
+    cy.getByDT('modal').within(() => {
+      cy.getByDT('payRecord').find('tbody').children().should('have.length', 2)
+      cy.getByDT('payRow').eq(0).find('input[data-test="amount"]').should('have.value', '71.43')
+      cy.getByDT('payRow').eq(0).find('label[data-test="check"]').click()
+
+      cy.get('button[type="submit"]').click()
+      cy.getByDT('successClose').click()
+      cy.getByDT('closeModal').should('not.exist')
+    })
+    assertPaymentsTabs(['Todo1', 'Sent1'])
+
+    // TODO - assert user2 received the payment.
+  })
+
+  it('user1 sends to user3 $100 (partial)', () => {
+    cy.getByDT('recordPayment').click()
+    cy.getByDT('modal').within(() => {
+      cy.getByDT('payRecord').find('tbody').children().should('have.length', 1)
+      cy.getByDT('payRow').eq(0).find('input[data-test="amount"]').should('have.value', '178.57')
+      cy.getByDT('payRow').eq(0).find('input[data-test="amount"]').clear({ force: true }).type('100')
+      cy.getByDT('payRow').eq(0).find('label[data-test="check"] input').should('be.checked')
+
+      cy.get('button[type="submit"]').click()
+      cy.getByDT('successClose').click()
+      cy.getByDT('closeModal').should('not.exist')
+    })
+
+    assertPaymentsTabs(['Todo1', 'Sent2'])
+
+    cy.getByDT('payList').within(() => {
+      cy.getByDT('payRow').eq(0).find('td:nth-child(2)').should('contain', '$78.57 out of $178.57')
+    })
+
+    // TODO - assert user3 received the payment.
+  })
+
+  it('one month later, user1 sends to user3 the missing $78.57', () => {
+    cy.visit('/')
+    cy.tick(60000 * 60 * 24 * 31) // time travel 1 month later
+    cy.getByDT('paymentsLink').click()
+    cy.get('[data-test-date]').should('have.attr', 'data-test-date', 'Mar 12')
+
+    assertPaymentsTabs(['Todo3', 'Sent2'])
+    // TODO: Assert each payment data. 1 late (Feb 29) and 2 new (Mar 31).
+
+    cy.getByDT('recordPayment').click()
+    cy.getByDT('modal').within(() => {
+      cy.getByDT('payRecord').find('tbody').children().should('have.length', 3)
+      cy.log('Verify and tick the late payment - user2')
+      cy.getByDT('payRow').eq(0).find('input[data-test="amount"]').should('have.value', '78.57')
+      // cy.getByDT('payRow').eq(0).find('[data-test="isPartial"]').should('exist') // TODO this
+      cy.getByDT('payRow').eq(0).find('[data-test="isLate"]').should('exist')
+      cy.getByDT('payRow').eq(0).find('label[data-test="check"]').click()
+
+      cy.log('Record one payment')
+      cy.get('button[type="submit"]').click()
+      cy.getByDT('successClose').click()
+      cy.getByDT('closeModal').should('not.exist')
+    })
+
+    // BUG - Making a late payment does not work as expected.
+    // Expected: Remove user3's late payment of $78.57.
+    // Reality: The late payment persists and the current month payment is reduced to partial ($100 out of $178.57)
+    // Once that is fixed, uncomment the assertions bellow:
```

Comment:
@taoeffect I started doing the tests for payments and it worked fine for normal payments. But with late-payments, some part is still broken. The list of payments is correct but the code to do the actual payment is broken. So I think this should be left for another PR.

Reply:
Well the commented out section above needs to be uncommented. If you want to comment out the late payment *tests* that's fine.

Reply:
It cannot be commented out because the code is still broken and it would fail the tests. That's for another PR (I'll update #918 to address that bug)